test/e2e/operator_test.go

```diff
@@ -670,3 +672,112 @@ func TestHostNetworkEndpointPublishingStrategy(t *testing.T) {
 		t.Fatalf("WARNING: cloud resources may have been leaked! failed to delete ingresscontroller %s: %v", name, err)
 	}
 }
+
+// TestInternalLoadBalancer creates an ingresscontroller with the
+// "LoadBalancerService" endpoint publishing strategy type with scope set to
+// "Internal" and verifies that the operator creates a load balancer and that
+// the load balancer has a private IP address.
+func TestInternalLoadBalancer(t *testing.T) {
+	cl, ns, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	infraConfig := &configv1.Infrastructure{}
+	err = cl.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig)
+	if err != nil {
+		t.Fatalf("failed to get infrastructure config: %v", err)
+	}
+
+	switch infraConfig.Status.Platform {
+	case configv1.AWSPlatformType:
+	case configv1.AzurePlatformType:
+	case configv1.BareMetalPlatformType:
+		t.Skip("test skipped on bare metal")
+	case configv1.GCPPlatformType:
+	case configv1.LibvirtPlatformType:
+		t.Skip("test skipped on libvirt")
+	case configv1.OpenStackPlatformType:
+	case configv1.NonePlatformType:
+		t.Skip("test skipped on \"none\" platform type")
+	case configv1.VSpherePlatformType:
+		t.Skip("test skipped on vSphere")
+	}
+
+	dnsConfig := &configv1.DNS{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, dnsConfig); err != nil {
+		t.Fatalf("failed to get DNS 'cluster': %v", err)
+	}
+
+	name := types.NamespacedName{Namespace: ns, Name: ingressControllerName}
+	domain := name.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newIngressController(name.Name, name.Namespace, domain, operatorv1.LoadBalancerServiceStrategyType)
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer = &operatorv1.LoadBalancerStrategy{
+		Scope: operatorv1.InternalLoadBalancer,
+	}
+	if err := cl.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller %s: %v", name, err)
+	}
+
+	// Wait for the load balancer and DNS to be ready.
+	err = waitForIngressControllerCondition(cl, 5*time.Minute, name, defaultAvailableConditions...)
+	if err != nil {
+		t.Errorf("failed to observe expected conditions: %v", err)
+	}
+
+	// Wait for the service to have at least one ingress.
+	lbService := &corev1.Service{}
+	err = wait.PollImmediate(1*time.Second, 2*time.Minute, func() (bool, error) {
```

Comment:
Isn't this covered by waitForIngressControllerCondition?

Reply:
I need the `lbService` so I can look at its status though.  Are you saying I should omit the polling loop around the `Get`?

Reply:
>Are you saying I should omit the polling loop around the Get?

Yeah, sorry, hopefully more clearly: you can just `Get` the service and error if not found, because the preceding higher-layer waitForIngressControllerCondition checks. If the condition check succeeded but the lb service doesn't exist, failing fast seems easy, an we'd want to investigate how that was possible